kafka-client/src/main/java/dev/responsive/kafka/api/async/internals/queues/KeyOrderPreservingQueue.java

```diff
@@ -0,0 +1,218 @@
+/*
```

Comment:
this is just `SchedulingQueue`, renamed so I could have a `SchedulingQueue` interface